ozaria/site/components/interactive/PageInteractive/draggableOrdering/index.vue

```diff
@@ -0,0 +1,160 @@
+<script>
+  import BaseInteractiveTitle from '../common/BaseInteractiveTitle'
+  import OrderingSlots from './OrderingSlots'
+  import Sortable from 'sortablejs'
+
+  export default {
+    components: {
+      'base-interactive-title': BaseInteractiveTitle,
+      'ordering-slots': OrderingSlots
+    },
+
+    props: {
+      interactive: {
+        type: Object,
+        required: true
+      },
+
+      introLevelId: {
+        type: String,
+        required: true
+      },
+
+      interactiveSession: {
+        type: Object,
+        default: undefined
+      },
+
+      codeLanguage: {
+        type: String
+      }
+    },
+
+    data () {
+      return {
+        draggableGroup: Math.random().toString(),
+
+        promptSlots: [
+          { id: 1, text: 'one option' },
+          { id: 2, text: 'two option' },
+          { id: 3, text: 'three option' },
+          { id: 4, text: 'four option' }
+        ],
+
+        answerSlots: (new Array(4)).fill(undefined)
+      }
+    },
+
+    mounted () {
+      const draggableUl = this.$refs['draggable-col']
+      Sortable.create(draggableUl, {
+        swap: true,
+        onChange: e => {
+          const temp = this.promptSlots[e.oldIndex]
+          this.promptSlots[e.oldIndex] = this.promptSlots[e.newIndex]
+          this.promptSlots[e.newIndex] = temp
+        }
+      })
+    }
+  }
+</script>
+
+<template>
+  <div class="draggable-ordering-container">
+    <base-interactive-title
+      :interactive="interactive"
+    />
+
+    <div class="prompt-row">
+      <div id='draggable-col' class='slots-container' key="draggable-col" ref="draggable-col">
+        <ul v-for="prompt in promptSlots" class="draggable-slot" :key="prompt.id">
+          <li>{{ prompt.text }}</li>
+        </ul>
+      </div>
+
+      <ordering-slots
+        v-model="answerSlots"
```

Comment:
The labels(answerSlots in this case) do not need to be draggable anymore. I tested this and it seems I can still drag column 1 to column 2 and then it leads to some error. These can be plain divs instead of ordering slots. Its fine if thats intentional and will be handled in a separate PR.

Reply:
I am not able to repro dragging from column 1 to column 2. But I can make them divs in the next PR to completely remove the possibility! Thanks for the comment.

Reply:
Oh thats strange, lets test it together tomorrow morning.